fl4health/losses/contrastive_loss.py

```diff
@@ -0,0 +1,65 @@
+import torch
+import torch.nn as nn
+
+
+class ContrastiveLoss(nn.Module):
+    def __init__(
+        self,
+        device: torch.device,
+        temperature: float = 0.5,
+    ) -> None:
+        """
+        This contrastive loss is implemented based on https://github.com/QinbinLi/MOON.
+        Contrastive loss aims to enhance the similarity between the features and their positive pairs
+        while reducing the similarity between the features and their negative pairs.
+        Args:
+            device (torch.device): device to use for computation
+            temperature (float): temperature to scale the logits
+        """
+
+        super().__init__()
+        self.device = device
+        self.temperature = temperature
+        self.cos_sim = torch.nn.CosineSimilarity(dim=-1).to(self.device)
+        self.ce_criterion = torch.nn.CrossEntropyLoss().to(self.device)
+
+    def forward(
+        self, features: torch.Tensor, positive_pairs: torch.Tensor, negative_pairs: torch.Tensor
+    ) -> torch.Tensor:
+        """
+        Compute the contrastive loss based on the features, positive pair and negative pairs. While every feature
+        has a positive pair, it can have multiple negative pairs. The loss is computed based on the similarity
+        between the feature and its positive pair.
```

Comment:
nit: similarity between the feature and its positive pair relative to negative pairs

Reply:
👍